client/src/pages/Home.tsx

```diff
@@ -1,16 +1,46 @@
+import Index from 'flexsearch';
 import { useEffect, useState } from 'react';
 import { useSearchParams } from 'react-router-dom';
 import { toast } from 'sonner';
 
+import data from '../../../seed/courses-2023-2024.json';
 import { CourseSearchBar } from '../components/CourseSearchBar';
 import { Layout } from '../components/Layout';
-import { repo } from '../lib/repo';
+import { dedupeArray } from '../lib/utils';
+import { Course } from '../model/Course';
+import { Instructor } from '../model/Instructor';
 import type { SearchResults } from '../model/SearchResults';
 
 const alerts: Map<string, string> = new Map([
   ['invalidMail', 'Please use a McGill email address to authenticate.'],
 ]);
 
+const courses: Course[] = data as Course[];
+const instructors: Instructor[] = dedupeArray(
+  courses.flatMap((course) => course.instructors),
+  (instructor) => instructor.name
+);
+
+const coursesIndex = new Index({
+  tokenize: 'forward',
+  limit: 4,
+});
+
+const instructorsIndex = new Index({
+  tokenize: 'forward',
+  limit: 2,
+});
+
+courses.forEach((course, i) =>
+  coursesIndex.add(
+    i,
+    `${course._id} ${course.subject} ${course.title} ${course.code}`
+  )
+);
+instructors.forEach((instructor, i) =>
+  instructorsIndex.add(i, instructor.name)
+);
```

Comment:
don't you need to update the course search bar in the header as well? just put this in some file under `/lib`